src/librustc_trans/type_.rs

```diff
@@ -140,6 +140,10 @@ impl Type {
         }
     }
 
+    pub fn c_int(ccx: &CrateContext) -> Type {
+        Type::i32(ccx)
```

Comment:
This doesn’t seem like a particularly good implementation of `c_int`. That being said, liblibc defines `type c_int = i32` for everything, so it is fine, I guess?

Reply:
Yea, my thoughts exactly. It should be derived from the target context, as done for Type::isize() with target_pointer_width. But I thought that's not needed yet, because libc currently always uses i32 anyways. (does rustc support target platforms with 16bit/64bit C int?)

Reply:
AVR uses a 16-bit `int`. I'll leave a comment on the tracking issue so this gets fixed if/when that gets upstreamed.

Reply:
And MSP430 also. We currently support that target, but I don't think that needs to block this PR since this is strictly better than what we're already doing. (using `i64`).

Reply:
`isize` is 16 bit on MSP: https://github.com/rust-lang/rust/blob/b7041bfab3a83702a8026fb7a18d8ea7d54cc648/src/librustc_back/target/msp430_none_elf.rs#L18

Reply:
Ah, I missed that. Going to work on a fix.